### PR TITLE
add weight in tree model output

### LINKED ISSUE #2037: Saving weight values of a node
Hi, right now for lightgbm trees following information is stored
1.Node index
2.Leaf count
3.Splitting variable
4.Threshold
5.Internal value
6.Leaf value
Is it possible to store the sum of weights too? It would be a great help in debugging or understanding the prediction.

### PR DESCRIPTION
**Pull request by guolinke**
fix #2037

### CODE REVIEW

File: src/boosting/gbdt_model_text.cpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 namespace LightGBM {
 
-const std::string kModelVersion = "v2";
+const std::string kModelVersion = "v3";
```
Comment (StrikerRUS) [inline src/boosting/gbdt_model_text.cpp]: Is new model format backward compatible with current v2?

**Comment by StrikerRUS** [comment]
@guolinke I can update plotting part if you don't mind.
https://github.com/microsoft/LightGBM/blob/611cf5d414a8273dfdd261651e2f127322a3db89/python-package/lightgbm/plotting.py#L392-L396

File: src/boosting/gbdt_model_text.cpp (hunk 1 above)
Comment (guolinke) [reply src/boosting/gbdt_model_text.cpp]: Yeah, it can load the v2 model file

**Comment by guolinke** [comment]
@StrikerRUS sure, thanks very much!

**Comment by StrikerRUS** [comment]
@guolinke Just updated plotting module in Python in the latest commit.

**Comment by StrikerRUS** [comment]
ping @jameslamb and @Laurae2 for R-package updates

https://github.com/microsoft/LightGBM/blob/master/R-package/R/lgb.model.dt.tree.R
https://github.com/microsoft/LightGBM/blob/master/R-package/R/lgb.interprete.R

**Comment by guolinke** [comment]
@StrikerRUS could this PR merge first? then a new PR for R package could be proposed.

**Comment by StrikerRUS** [comment]
@guolinke Yeah, sure!

**Comment by jameslamb** [comment]
Created #2281 to track the R work, agree with the decision that was made to merge this and come back to R.

**Pull request merged**